benchmarks/optimize.py

```diff
@@ -0,0 +1,70 @@
+from typing import cast
+from typing import List
+from typing import Union
+
+import optuna
+from optuna.samplers import BaseSampler
+from optuna.samplers import CmaEsSampler
+from optuna.samplers import RandomSampler
+from optuna.samplers import TPESampler
+from optuna.testing.storage import StorageSupplier
+
+
+def parse_args(args: str) -> List[Union[int, str]]:
+    ret: List[Union[int, str]] = []
+    for arg in map(lambda s: s.strip(), args.split(",")):
+        try:
+            ret.append(int(arg))
+        except ValueError:
+            ret.append(arg)
+    return ret
+
+
+SAMPLER_MODES = [
+    "random",
+    "tpe",
+    "cmaes",
+]
+
+
+def create_sampler(sampler_mode: str) -> BaseSampler:
+    if sampler_mode == "random":
+        return RandomSampler()
+    elif sampler_mode == "tpe":
+        return TPESampler()
+    elif sampler_mode == "cmaes":
+        return CmaEsSampler()
+    else:
+        assert False
+
+
+class TimeOptimize:
```

Comment:
I'd say naming classes like `XxxSuite` is more common.

https://github.com/scikit-image/scikit-image/tree/main/benchmarks
https://asv.readthedocs.io/en/stable/writing_benchmarks.html

```suggestion
class OptimizeSuite:
```